api/types/okta.go

```diff
@@ -180,6 +181,8 @@ type OktaAssignment interface {
 	GetUser() string
 	// GetActions will return the list of actions that will be performed as part of this assignment.
 	GetActions() []OktaAssignmentAction
+	// GetCleanupTime will return the optional time that the assignment should be cleaned up.
+	GetCleanupTime() *time.Time
```

Comment:
Any reason this is a pointer? Are you using it being `nil` to differentiate states (not set, set as zero time, set at a particular time)?

Reply:
This field is optional, so nil is being used to determine if it's not set.

Reply:
Nit: technically you can make it non-pointer and use `IsZero()` to determine whether it's set or not but I don't have a strong opinion on this one.

Reply:
Feel the same way as Roman on this one.